core/commands/storage.go

```diff
@@ -1294,7 +1294,7 @@ This command print upload and payment status by the time queried.`,
 		// check and get session info from sessionMap
 		ssID := req.Arguments[0]
 		n, err := cmdenv.GetNode(env)
-		ss, err := storage.GlobalSession.GetSession(n, storage.RenterStoragePrefix, ssID)
+		ss, err := storage.GlobalSession.GetSessionWithoutLock(n, storage.RenterStoragePrefix, ssID)
```

Comment:
So we don't really know why certain routines hold the locks for too long?

Reply:
> So we don't really know why certain routines hold the locks for too long?

actually it's mostly stuck at `send challenge questions` step.
typically when we upload a large file say 250M+.

Reply:
at the meantime, `upload status` is blocked by mutex.Lock().